samples/snippets/fsharp/VS_Snippets_CLR_System/system.aggregateexception.class/fs/exception1.fs

```diff
@@ -0,0 +1,56 @@
+﻿// <Snippet1>
+open System
+open System.IO
+open System.Threading.Tasks
+
+let getAllFiles str =
+    // Should throw an UnauthorizedAccessException exception.
+    System.IO.Directory.GetFiles(str, "*.txt", System.IO.SearchOption.AllDirectories)
+
+// Get a folder path whose directories should throw an UnauthorizedAccessException.
+let path =
+    Environment.SpecialFolder.UserProfile
+    |> Environment.GetFolderPath
+    |> Directory.GetParent
+    |> fun x -> x.FullName
```

Comment:
In general please don't use `|> fun x -> ...` in beginner code snippets.  The style is OK but confusing for beginners.  Just use a let binding, e.g. 

```fsharp
let path =
    let parent =
        Environment.SpecialFolder.UserProfile
        |> Environment.GetFolderPath
        |> Directory.GetParent
    parent.FullName
```